src/pages/PrimeCalculator/Form/index.tsx

```diff
@@ -14,75 +20,137 @@ import {
   Spinner,
   TokenIconWithSymbol,
 } from 'components';
+import FunctionKey from 'constants/functionKey';
+import useDebounceValue from 'hooks/useDebounceValue';
 import { useGetToken } from 'packages/tokens';
 import { useTranslation } from 'packages/translations';
-import { Token } from 'types';
-import { areAddressesEqual } from 'utilities';
+import { useAccountAddress, useChainId } from 'packages/wallet';
+import { Asset } from 'types';
+import {
+  areAddressesEqual,
+  convertMantissaToTokens,
+  formatCentsToReadableValue,
+  formatTokensToReadableValue,
+} from 'utilities';
 
 import TEST_IDS from '../testIds';
 import { Field } from './Field';
 import { RewardDetails } from './RewardDetails';
+import { getInitialValues } from './getInitialValues';
 
 export const QUERY_PARAM_TOKEN_ADDRESS = 'tokenAddress';
 
-const formSchema = z.object({
-  stakedAmountXvs: z.string().min(1),
-  suppliedAmountTokens: z.string().min(1),
-  borrowedAmountTokens: z.string().min(1),
-});
+const validateNumericString = (str: string, minValue = 0) => {
+  if (!str) {
+    return false;
+  }
+  const n = Number(str);
+  const isNumber = !Number.isNaN(n);
+  const gteMinValue = minValue ? n >= minValue : true;
+
+  return isNumber && gteMinValue;
+};
 
 export const Form: React.FC = () => {
+  const { chainId } = useChainId();
   const { t } = useTranslation();
-  const { data: getLegacyPoolData, isLoading: isGetLegacyPoolLoading } = useGetLegacyPool();
+  const xvs = useGetToken({
+    symbol: 'XVS',
+  });
+  const { accountAddress } = useAccountAddress();
+  const { data: getLegacyPoolData, isLoading: isGetLegacyPoolLoading } = useGetLegacyPool(
+    {
+      accountAddress,
+    },
+    {
+      refetchInterval: false,
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      staleTime: Infinity,
+      queryKey: [FunctionKey.GET_LEGACY_POOL, { accountAddress, chainId }, 'PrimeCalculator'],
+    },
+  );
+  const { data: getPrimeStatusData, isLoading: isGetPrimeStatusLoading } = useGetPrimeStatus({
+    accountAddress,
+  });
+  const primeMinimumStakedXvsMantissa = getPrimeStatusData?.primeMinimumStakedXvsMantissa;
+  const primeMaximumStakedXvsMantissa = getPrimeStatusData?.primeMaximumStakedXvsMantissa;
+  const xvsVaultPoolIndex = getPrimeStatusData?.xvsVaultPoolId;
```

Comment:
This query will be easier to work with if it always returns the correct shape of an object or throws an error. If the values can be undefined they should be returned undefined. Then you can clean this up by destructuring instead of manually reassigning to variables
```suggestion
  const { data:{primeMinimumStakedXvsMantissa, primeMaximumStakedXvsMantissa, xvsVaultPoolId}, isLoading: isGetPrimeStatusLoading } = useGetPrimeStatus({
    accountAddress,
  });
  const primeMinimumStakedXvsMantissa = getPrimeStatusData?.primeMinimumStakedXvsMantissa;
  const primeMaximumStakedXvsMantissa = getPrimeStatusData?.primeMaximumStakedXvsMantissa;
  const xvsVaultPoolIndex = getPrimeStatusData?.xvsVaultPoolId;
```